jwcrypto/jwe.py

```diff
@@ -542,6 +600,25 @@ def _get_alg_enc_from_headers(self, jh):
         enc = self._jwa(encname)
         return alg, enc
 
+    def _encrypt(self, alg, enc, jh):
+        aad = base64url_encode(self.objects.get('protected', ''))
+        if 'aad' in self.objects:
+            aad += '.' + base64url_encode(self.objects['aad'])
```

Comment:
Nope self.objects['add'] is in binary form throughout the code, it is not encoded on __init__ and it is decoded on deserialization.